### PR TITLE
Rename nodes to nodeSets, nodeCount to count

### LINKED ISSUE #1488: Should we rename nodes in Elasticsearch spec
This came up in a [comment](https://github.com/elastic/cloud-on-k8s/pull/1463#discussion_r310187518) for #1463. 

I got a feeling that current name ("nodes") implies that it holds a list of nodes, but that's not accurate and might be misleading. It does hold a list, but each item describes a set of identical es nodes and holds:
- name for the set,
- es config to use for all nodes in the set,
- number of nodes in the set,
- pod template to use for all nodes in the set,
- volume claim templates to use for all nodes in the set.

I don't have full context as this was before my time, but seems like this field was renamed already from topologies to topology and then to nodes. 

Alternatives that came up: nodeGroups, nodeSets, nodeSpecs, nodeTemplates.

I don't really feel strongly about this, but we decided to ask broader, hence calling @agup006 @jordansissel @Crazybus @zanbel @uric.

### PR DESCRIPTION
**Pull request by pebrc**
As it says in the title. I am unsure about the the `nodeCount` rename for Kibana and APM but thought we should aim for consistency. 

Fixes #1488

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -8,35 +8,6 @@ Third party libraries used by the Elastic Cloud on Kubernetes project
...
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-THE SOFTWARE.
-
-
```
Comment (pebrc) [inline NOTICE.txt]: Not entirely sure where this is coming from

File: config/crds/apm.k8s.elastic.co_apmservers.yaml [hunk 2]
```diff
@@ -34,342 +34,6 @@ spec:
...
-            secretTokenSecret:
-              description: SecretTokenSecretName is the name of the Secret that contains
-                the secret token
-              type: string
-            service:
-              description: ExternalService is the name of the service the agents should
-                connect to.
-              type: string
-          type: object
-      type: object
```
Comment (pebrc) [inline config/crds/apm.k8s.elastic.co_apmservers.yaml]: Now that we introduce a difference between versions, the schema is gone for APM/Kibana too :-(

Comment (sebgl) [reply config/crds/apm.k8s.elastic.co_apmservers.yaml]: Well, at least that's consistent 😄

**Comment by thbkrkr** [comment]
I think there are still some `nodes:` and `nodeCount:` to rename in `docs/elasticsearch-spec.asciidoc`.
I also spotted that `nodeSets.name` is missing in the yaml snippet (can be done in a follow-up but it's the occasion to fix it).

File: docs/k8s-quickstart.asciidoc [hunk 3]
```diff
@@ -247,7 +248,8 @@ metadata:
 spec:
   version: {version}
   nodes:
```
Comment (thbkrkr) [inline docs/k8s-quickstart.asciidoc]:
```suggestion
  nodeSets:
```

File: pkg/apis/kibana/v1beta1/kibana_types.go [hunk 4]
```diff
@@ -24,8 +24,8 @@ type KibanaSpec struct {
 	// Image represents the docker image that will be used.
 	Image string `json:"image,omitempty"`
 
-	// NodeCount defines how many nodes the Kibana deployment must have.
-	NodeCount int32 `json:"nodeCount,omitempty"`
+	// Count defines how many nodes the Kibana deployment must have.
+	Count int32 `json:"nodeCount,omitempty"`
```
Comment (thbkrkr) [inline pkg/apis/kibana/v1beta1/kibana_types.go]:
```suggestion
	Count int32 `json:"count,omitempty"`
```

**Comment by pebrc** [comment]
Jenkins test this please

**Pull request merged**